2_compare_results.ipynb

```diff
@@ -0,0 +1,639 @@
+      "text/plain": [
+       "<Figure size 640x480 with 1 Axes>"
+      ]
+     },
+     "metadata": {},
+     "output_type": "display_data"
+    }
+   ],
+   "source": [
+    "sns.set_theme(context=\"talk\", palette=[\"#7400ff\", \"#e01073\"])\n",
+    "ax = sns.barplot(data=df)\n",
+    "for container in ax.containers:\n",
+    "    ax.bar_label(container=container, fontsize=11, fmt=\"%.1fs\", label_type=\"center\")\n",
+    "ax.set_ylabel(ylabel=\"Data load time per epoch\\n ◀ seconds, lower is better\")\n",
+    "ax.set_xlabel(\n",
+    "    xlabel=\"   (with GDS)            (without GDS) \\n\\n xarray backend engine\"\n",
+    ")\n",
+    "ax.set_title(label=\"Reading ERA5 data with/without GPUDirect Storage\")\n",
+    "fig = ax.get_figure()\n",
+    "fig.savefig(fname=\"figures/compare_kvikio_zarr.svg\", bbox_inches=\"tight\")"
```

Comment:
For reference, this is the results when running from my freshly booted system (cold start):

![compare_kvikio_zarr](https://github.com/weiji14/foss4g2023oceania/assets/23487320/ff7adcfd-f35c-4cba-9650-e4b1f7f988ba)

The `kvikIO` engine shows:
- Median: 12.1870 s
- Mean ± Standard deviation: 14.2480 ± 6.6434 seconds/epoch

The `zarr` engine shows:
- Median: 15.9633 seconds/epoch
- Mean ± Standard deviation: 16.0398 ± 0.4271 seconds/epoch

which is why I switched to reporting the Median time instead of the Mean time at 25d1642daf907cfe789932ab992bb1d395b05c0c.

Reply:
Run 2, after the GPU has warmed up a bit, and presumably there's some caching going on:

![compare_kvikio_zarr](https://github.com/weiji14/foss4g2023oceania/assets/23487320/cd17f64c-271b-475e-bb5b-afce0d266d08)

The kvikIO engine shows:
- Median: 11.9764 seconds/epoch
- Mean: 12.2756 ± 1.1680 seconds/epoch

The zarr engine shows:
- Median: 15.7782 seconds/epoch
- Mean: 15.8177 ± 0.2417 seconds/epoch

And I just realized, the bar plot is showing the mean value, not the median :sweat_smile:

Reply:
Run 3, after fixing the bar plot to show the median value instead of the mean value.

![compare_kvikio_zarr](https://github.com/weiji14/foss4g2023oceania/assets/23487320/7dd5cd16-5986-482a-883c-0366b21a3f61)

The `kvikIO` engine shows:

- Median: 11.8606 seconds/epoch
- Mean: 12.1578 ± 0.8311 seconds/epoch

The `zarr` engine shows:

- Median: 16.0240 seconds/epoch
- Mean: 15.9295 ± 0.3761 seconds/epoch

Gonna go with this one (commit at 6235109e7c0dbdcc1f463f384713e5a1578b87a0) since 16.0s and 11.9s are almost round numbers :slightly_smiling_face: